fix(transformer/typescript): remove properties with definite assignment assertion

Dunqing opened the pull request:
close: https://github.com/rolldown/tsdown/issues/399


Here is a [TypeScript Playground](https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBAYwDYEMDOa4FkCeBhVDOAbwCg44A7FAW2AEIAuONGKAS0oHNyXQmWbTjwC+pUrgLo0AOjBQIMRTjDAZ1OnAC8cAOQAvABYpuaE7on5Cs+YuWqZaUNr00USYBaA), which shows the expected output.

A Babel test failed because Babel's transforming behavior differs from TypeScript's. Additionally, ESBuild isn't the same.
See [Babel Playground](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&corejs=3.42&code_lz=MYGwhgzhAECC0G8BQ1oA8CEBuJBfJQA&sourceType=module&lineWrap=true&presets=env%2Ctypescript&version=7.28.2&externalPlugins=%40babel%2Fplugin-transform-class-properties%407.27.1)

Boshen commented:
### Merge activity

* **Aug 1, 11:45 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(transformer/typescript): remove properties with definite assignment assertion (#12713)

close: https://github.com/rolldown/tsdown/issues/399

Here is a [TypeScript Playground](https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBAYwDYEMDOa4FkCeBhVDOAbwCg44A7FAW2AEIAuONGKAS0oHNyXQmWbTjwC+pUrgLo0AOjBQIMRTjDAZ1OnAC8cAOQAvABYpuaE7on5Cs+YuWqZaUNr00USYBaA), which shows the expected output.
